### PR TITLE
Jc creation zombie

### PR DESCRIPTION
**Pull request by jdcanas**
\<your comments for this PR go here\>

Have you read [CONTRIBUTING.md](https://github.com/DataBiosphere/leonardo/blob/develop/CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've documented my API changes in Swagger

In all cases:

- [ ] Get a thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green
- [ ] Squash and merge; you can delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

**Comment by jdcanas** [comment]
jenkins retest

**Comment by jdcanas** [comment]
jenkins retest

File: src/test/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/ZombieClusterMonitorSpec.scala [hunk 1]
```diff
@@ -131,7 +131,7 @@ class ZombieClusterMonitorSpec extends TestKit(ActorSystem("leonardotest")) with
     }
   }
 
-  it should "should not zombify Creating clusters" in isolatedDbTest {
+  it should "should zombify creating cluster after hang period" in isolatedDbTest {
```
Comment (Qi77Qi) [inline src/test/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/ZombieClusterMonitorSpec.scala]: Do we need to keep the original test that will not zombify creating cluster if it's within hang period?

**Comment by jdcanas** [comment]
jenkins retest

File: src/test/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/ZombieClusterMonitorSpec.scala (hunk 1 above)
Comment (jdcanas) [reply src/test/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/ZombieClusterMonitorSpec.scala]: I was thinking the same thing on the first pass. Just added it and cleaned up the hardcoded constants

File: src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/ZombieClusterMonitor.scala [hunk 2]
```diff
@@ -95,9 +96,18 @@ class ZombieClusterMonitor(config: ZombieClusterConfig, gdDAO: GoogleDataprocDAO
...
-    // Clusters in Creating status may not yet exist in Google. Therefore treat all Creating clusters as active.
+    // Clusters in Creating status may not yet exist in Google. Clusters in creating (that are not hanging, as defined by config.creationHangTolerance) are considered active
     if (cluster.status == ClusterStatus.Creating) {
-      Future.successful(true)
+
+        val secondsSinceClusterCreation: Long = Duration.between(cluster.auditInfo.createdDate, Instant.now()).getSeconds
+
+        //determine if the creating cluster has been hanging past the tolerance threshold
+        if (secondsSinceClusterCreation > config.creationHangTolerance.toSeconds) {
+           Future.successful(false)
```
Comment (rtitle) [inline src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/ZombieClusterMonitor.scala]: Should this `if` block check the cluster status in Google? I'm thinking we want the following logic:

| status in Leo | present in Google | isClusterActiveInGoogle returns... |
| ------------- | ------------------| ---------------------------------|
| Creating, inside the hang tolerance | yes | true |
| Creating, inside the hang tolerance | no | true |
| Creating, outside the hang tolerance | yes | true |
| Creating, outside the hang tolerance | no | false |

File: src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/ZombieClusterMonitor.scala [hunk 3]
```diff
@@ -95,17 +96,16 @@ class ZombieClusterMonitor(config: ZombieClusterConfig, gdDAO: GoogleDataprocDAO
...
-        logger.warn(s"Unable to check status of cluster ${cluster.projectNameString} for zombie cluster detection", e)
-        true
-      }
+
+    val secondsSinceClusterCreation: Long = Duration.between(cluster.auditInfo.createdDate, Instant.now()).getSeconds
+    //this or'd with the google cluster status gives creating clusters a grace period before they are marked as zombies
+    val isWithinHangTolerance =  cluster.status == ClusterStatus.Creating && secondsSinceClusterCreation > config.creationHangTolerance.toSeconds
+
+   gdDAO.getClusterStatus(cluster.googleProject, cluster.clusterName) map { clusterStatus =>
+      (ClusterStatus.activeStatuses contains clusterStatus) || isWithinHangTolerance
```
Comment (Qi77Qi) [inline src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/ZombieClusterMonitor.scala]: I could be misreading, but I think this is not quite right. `ClusterStatus.activeStatuses` includes `Creating`, so we won't actually be checking `isWithinHangTolerance` when status is `Creating`.

Comment (jdcanas) [reply src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/ZombieClusterMonitor.scala]: As I understand it based, if the cluster is active in google, it is not considered a zombie in any scenario, as the point of the zombie monitor is to remove clusters in Leo DB that aren't present in google, and the functionality is not designed to extend the other way.

Thus, the hang tolerance is only relevant in the cases in which the cluster is not active in google. That being said, I think I should split line 102 into two separate variables so it is more clear though (after I merge this maybe, so I don't have to go through the automation tests again). Something like:

`val shouldConsiderCreatingClusterActive = isWithinHangTolerance && cluster.status == ClusterStatus.Creating`

Which should maintain:

| Creating  | In google | Within hang tolerance | Desired Result |
|---|---|---|---|
| Yes | Yes | Yes | true |
| Yes | No | Yes | true |
| Yes | Yes | No | true |
| Yes | No  | No | false |
| No | Yes | Any | true |
| No | No | Any | false |

Comment (Qi77Qi) [reply src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/ZombieClusterMonitor.scala]: thanks for the explanation. I think that makes sense. one nit, the method name is a bit misleading after we add the hangTolerance logic. `isClusterActive` would be better. A comment like what you said here would be helpful too.

**Pull request merged**